src/EditorFeatures/CSharpTest/InlineDeclaration/CSharpInlineDeclarationTests.cs

```diff
@@ -0,0 +1,347 @@
+@"class C
+{
+    void M(string v)
+    {
+        [|int|] i;
+        if (int.TryParse(v, out i))
+        {
+        } 
+    }
+}",
+@"class C
+{
+    void M(string v)
+    {
+        if (int.TryParse(v, out int i))
+        {
+        } 
+    }
+}", options: UseImplicitTypeTests.ImplicitTypeButKeepIntrinsics());
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineDeclaration)]
+        public async Task TestAvailableWhenWrittenAfter1()
+        {
+            await TestAsync(
+@"class C
+{
+    void M()
+    {
+        [|int|] i;
+        if (int.TryParse(v, out i))
+        {
+        }
+        i = 0;
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        if (int.TryParse(v, out int i))
+        {
+        }
+        i = 0;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineDeclaration)]
+        public async Task TestMissingWhenWrittenBetween1()
+        {
+            await TestMissingAsync(
+@"class C
+{
+    void M()
+    {
+        [|int|] i;
+        i = 0;
+        if (int.TryParse(v, out i))
+        {
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineDeclaration)]
+        public async Task TestMissingWithInitializer()
+        {
+            await TestMissingAsync(
+@"class C
+{
+    void M()
+    {
+        [|int|] i = 0;
+        if (int.TryParse(v, out i))
+        {
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineDeclaration)]
+        public async Task TestAvailableInOuterScopeIfNotWrittenOutside()
+        {
+            await TestMissingAsync(
+@"class C
+{
+    void M()
+    {
+        [|int|] i = 0;
+        {
+            if (int.TryParse(v, out i))
+            {
+            }
+            i = 1;
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineDeclaration)]
+        public async Task TestMissingIfWrittenAfterInOuterScope()
+        {
+            await TestMissingAsync(
+@"class C
+{
+    void M()
+    {
+        [|int|] i = 0;
+        {
+            if (int.TryParse(v, out i))
+            {
+            }
+        }
+        i = 1;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineDeclaration)]
+        public async Task TestMissingIfWrittenBetweenInOuterScope()
+        {
+            await TestMissingAsync(
+@"class C
+{
+    void M()
+    {
+        [|int|] i = 0;
+        {
+            i = 1;
+            if (int.TryParse(v, out i))
+            {
+            }
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineDeclaration)]
+        public async Task TestMissingInNonOut()
+        {
+            await TestMissingAsync(
+@"class C
+{
+    void M()
+    {
+        [|int|] i;
+        if (int.TryParse(v, i))
+        {
+        } 
+    }
+}");
```

Comment:
Also, I get that the diagnostic analyzer performs a syntactic check, but if that code ever changes, this test *could* let a bug slip through.